CONTRIBUTING.md

```diff
@@ -0,0 +1,29 @@
+# Contribution Guidelines
+
+### New features/Issues
+
+To suggest a new feature or report an issue please use the 
+[issue](https://github.com/UCL/TDMS/issues) board providing details of either 
+the feature or the bug with steps to reproduce it. To contribute to the code
+base please
+1. [fork](https://docs.github.com/en/get-started/quickstart/fork-a-repo)
+this repository,
+2. commit and push the changes to your fork
+3. and submit a [pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) (PR) against the _main_ branch, being mindful of the guidelines below.
+
+
+### Pull requests
+
+PRs should follow the following guidelines
+
+- Contain a **context** or **description** of the change
+- Include any useful hints for reviewers
+- Not be excessively large (500 line diff at a maximum)
```

Comment:
This is a good suggestion, but I'm not convinced about putting down a threshold like that.  There may be legitimate cases (for example when refactoring) where you may end up touching more than 500 lines.  How about something along the lines of "keep PRs to a single logically consistent set of changes".  I'm not fond of the wording, but the point is to avoid cramming multiple unrelated changes in a single PR, which I think is the more important principle to follow.

Reply:
I mostly agree with this. I guess it depends on the type of refactoring but I'd still be in favour of never having to look at >500 line diffs. Personal preference I guess 🤷🏼 
I'll tone it down appropriately